File: backend/src/main/java/com/neon1024/backend/models/Image.java
package com.neon1024.backend.models;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.persistence.GenerationType;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.Max;

import java.util.UUID;

@Entity
@Table(name = "Images")
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;
    @NotNull
    private String publicId;
    @NotNull
    private String url;
    @NotNull
    @Min(value = 0)
    @Max(value = 7)
    private Integer position;

    public UUID getId() {
        return this.id;
    }

    public Product getProduct() {
        return this.product;
    }

    public void setProductId(Product product) {
        this.product = product;
    }

    public String getPublicId() {
        return this.publicId;
    }

    public void setPublicId(String publicId) {
        this.publicId = publicId;
    }

    public String getUrl() {
        return this.url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Integer getPosition() {
        return this.position;
    }

    public void setPosition(Integer position) {
        this.position = position;
    }
}

File: frontend/src/models/Image.ts
class Image {
    private id: string;
    private productId: string;
    private publicId: string;
    private url: string;
    private position: number;

    public constructor(
        id: string,
        productId: string,
        publicId: string,
        url: string,
        position: number
    ) {
        this.id = id;
        this.productId = productId;
        this.publicId = publicId;
        this.url = url;
        this.position = position;
    }

    public getId(): string {
        return this.id;
    }

    public setId(id: string) {
        this.id = id;
    }

    public getProductId(): string {
        return this.productId;
    }

    public setProductId(productId: string) {
        this.productId = productId;
    }

    public getPublicId(): string {
        return this.publicId;
    }

    public setPublicId(publicId: string) {
        this.publicId = publicId;
    }

    public getUrl(): string {
        return this.url;
    }

    public setUrl(url: string) {
        this.url = url;
    }

    public getPosition(): number {
        return this.position;
    }

    public setPosition(position: number) {
        this.position = position;
    }
}

export default Image;
